app/src/main/java/com/kickstarter/ui/activities/AccountActivity.kt

```diff
@@ -143,8 +163,8 @@ class AccountActivity : BaseActivity<AccountViewModel.ViewModel>() {
             override fun onNothingSelected(parent: AdapterView<*>?) {}
 
             override fun onItemSelected(parent: AdapterView<*>?, view: View?, postion: Int, id: Long) {
-                if (currentCurrencySelection != null && currentCurrencySelection!!.ordinal != postion) {
-                    newCurrencySelection = CurrencyCode.values()[postion]
+                if (currentCurrencySelection != null && supportedCurrencies.indexOf(currentCurrencySelection!!) != postion) {
```

Comment:
Can you unwrap the currentCurrencySelection, so we don't need to use that forcing non optional `!!`
`currentCurrencySelection?.let {
                    if (supportedCurrencies.indexOf(it) != postion) {
                        newCurrencySelection = supportedCurrencies[postion]
                        lazyFollowingOptOutConfirmationDialog().show()
                    }
                }`